analyzers/src/SonarAnalyzer.Common/Rules/ClassShouldNotBeEmptyBase.cs

```diff
@@ -79,5 +79,8 @@ public abstract class ClassShouldNotBeEmptyBase<TSyntaxKind, TDeclarationSyntax>
 
     private static bool HasNonPublicDefaultConstructor(TDeclarationSyntax declaration, SemanticModel model) =>
         model.GetDeclaredSymbol(declaration) is INamedTypeSymbol classSymbol
-        && classSymbol.BaseType.Constructors.FirstOrDefault(x => x.Parameters.Length == 0) is { DeclaredAccessibility: not Accessibility.Public };
+        && classSymbol.BaseType.Constructors.FirstOrDefault(x => x.Parameters.Length == 0) is { } constructor
+        && (constructor.GetEffectiveAccessibility() < classSymbol.DeclaredAccessibility
+            // GetEffectiveAccessibility is not handling Protected
+            || (constructor.DeclaredAccessibility == Accessibility.Protected && classSymbol.DeclaredAccessibility > Accessibility.Protected));
```

Comment:
Why is this one not in the `GetEffectiveAccessibility` method?

Reply:
But our implementation of `GetEffectiveAccessibility`. The reason that `protected` is ignored is most likely, that `protected` can be considered more accessible than `internal`, depending on your point of view:
```cs
public virtual class
{
    internal int NotAccessibleFromOtherAssemblies { get; }
    protected int AccessibleFromOtherAssembliesWhenInheritingTheClass { get; }
}
```